update/storing/hardforkStorer.go

```diff
@@ -0,0 +1,143 @@
+package storing
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/batch"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/update"
+)
+
+var log = logger.GetOrCreate("update/storing")
+
+// ArgHardforkStorer represents the argument for the hardfork storer
+type ArgHardforkStorer struct {
+	KeysStore   storage.Storer
+	KeyValue    storage.Storer
+	Marshalizer marshal.Marshalizer
+}
+
+type hardforkStorer struct {
+	keysStore   storage.Storer
+	keyValue    storage.Storer
+	marshalizer marshal.Marshalizer
+
+	mut  sync.Mutex
+	keys map[string][][]byte
+}
+
+// NewHardforkStorer returns a new instance of a specialized storer used in the hardfork process
+func NewHardforkStorer(arg ArgHardforkStorer) (*hardforkStorer, error) {
+	if check.IfNil(arg.KeysStore) {
+		return nil, fmt.Errorf("%w for keys", update.ErrNilStorage)
+	}
+	if check.IfNil(arg.KeyValue) {
+		return nil, fmt.Errorf("%w for key-values", update.ErrNilStorage)
+	}
+	if check.IfNil(arg.Marshalizer) {
+		return nil, update.ErrNilMarshalizer
+	}
+
+	return &hardforkStorer{
+		keysStore:   arg.KeysStore,
+		keyValue:    arg.KeyValue,
+		marshalizer: arg.Marshalizer,
+		keys:        make(map[string][][]byte),
+	}, nil
+}
+
+// Write adds the pair (key, value) in the state storer. Also, it does record the connection between the identifier and
+// the key
+func (hs *hardforkStorer) Write(identifier string, key []byte, value []byte) error {
+	hs.mut.Lock()
+	defer hs.mut.Unlock()
+
+	hs.keys[identifier] = append(hs.keys[identifier], key)
```

Comment:
could this get too large for some identifiers? transactions?

Reply:
that's only the hashes, I suspect not to get too large

Reply:
ok